cmd/log-shuttle/main.go

```diff
@@ -244,8 +244,9 @@ func main() {
 
 	go LogFmtMetricsEmitter(s.MetricsRegistry, config.StatsSource, config.StatsInterval, s.Logger)
 
-	// Blocks until os.Stdin is closed
+	// Blocks until os.Stdin errors
 	s.ReadLogLines(os.Stdin)
```

Comment:
That means you don't care about the errors. CloseReaders will return any errors reported by Close() for each of the readers, but and actual read errors are discarded. FWIW: This is the dogwood use case.